docs/writing/posts/stochastic-software.md

```diff
@@ -0,0 +1,114 @@
+
+Here are some examples of recommendations that we can make based on this data:
+
+1. Our performance in internet searches is clearly underwhelming, but the count is quite low.
+2. In the meantime, we can disable this feature, knowing that it won't significantly impact our users.
+3. Personal data queries have a very high volume, but the quality is lacking.
+4. We should focus on building experiments that improve the quality of personal data queries.
+5. Since we can't run a backtest on users thumbs up and thumbs down ratings, we should consider a different metric like embedding reranking scores
+6. If we can show that changing our retrieval system can improve re-ranking scores, we should go and verify whether or not re-ranking scores correlate with quality and be able to iterate confidently knowing that we might be able to improve the final outcome.
+
+## Negative results are still results
+
+We're not in academia. A negative result is still a result. The goal isn't to publish novel research, the goal is to figure out how to prioritize our limited resources. Remember that to make a decision is to cut off. If we get a negative result or a neutral result, then the outcome is the same, we have made a decision. We have made a decision to cut off this line of inquiry, maybe not forever, but at least for now.
+
+It's also important to trust your judgment too. Just because we're going to cut off this line of reasoning now, it's still good to write up a little memo, explain what happened, write down other things we may not have considered to answer the question, "Under what conditions will we revisit this line of inquiry?"
+
+## Art vs. science
+
+Many people transitioning from classical software engineering to machine learning are often surprised by the empirical nature of the results we obtain. Instead of always having unit tests, we are sampling from the distribution of potential inputs and building a internal model how this system operatess.
+
+I hope that this article has helped you understand the importance of focusing on outcomes, metrics, and experiments instead of trying to think our way through edge cases and long tails. Additionally, I encourage you to develop the habit of making decisions and eliminating other possibilities. Lastly, I hope you will cultivate the practice of documenting your results and sharing them with your team, fostering a collective learning experience.
+
+As you're building these probabilistic systems ask youself:
+
+1. Would I make more progress if I just set up an experiment?
+2. Do I know what I want to measure and improve when I set up this experiment?
+3. What is the decision that I'm going to make as a result of seeing the results of this experiment?
+4. What are the conditions under which I will revisit this if I see neutral or negative results?
+5. Can I update my mental model of the system based on the results of this experiment, to plan future work?
```

Comment:
```suggestion
As you're building these probabilistic systems, ask yourself:

1. Would I make more progress if I just set up an experiment?
2. Do I know what I want to measure and improve when I set up this experiment?
3. What is the decision that I'm going to make as a result of seeing the results of this experiment?
4. What are the conditions under which I will revisit this if I see neutral or negative results?
5. Can I update my mental model of the system based on the results of this experiment to plan future work?

I hope that this article has helped you understand the importance of focusing on outcomes, metrics, and experiments instead of trying to think our way through edge cases and long tails. Additionally, I encourage you to develop the habit of making decisions and eliminating other possibilities. Lastly, I hope you will cultivate the practice of documenting your results and sharing them with your team, fostering a collective learning experience.

```